plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -459,7 +459,15 @@ private Document getTableMetadata(String schemaName, String tableName)
                 Document metadata = new Document(TABLE_NAME_KEY, tableName);
                 metadata.append(FIELDS_KEY, guessTableFields(schemaName, tableName));
 
-                schema.createIndex(new Document(TABLE_NAME_KEY, 1), new IndexOptions().unique(true));
+                boolean isSchemaCollectionIndexed = getIndexes(schemaName, tableName).stream()
+                        .anyMatch(mongoIndex -> mongoIndex.getKeys().stream()
+                                .anyMatch(mongodbIndexKey -> TABLE_NAME_KEY.equals(mongodbIndexKey.getName()))
```

Comment:
This logic may return wrong result in case of compound index. I would change as subsequent comment.